app/src/main/java/com/codingblocks/cbonlineapp/activities/HomeActivity.kt

```diff
@@ -146,12 +147,27 @@ class HomeActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelecte
             createShortcut()
         }
         nav_view.getHeaderView(0).login_button.setOnClickListener {
+            val builder = AlertDialog.Builder(this)
+            val dialogView=layoutInflater.inflate(R.layout.logout_dialogue , null)
+            builder.setView(dialogView)
+            builder.setTitle("Confirm Logout")
+            builder.setMessage("Are you sure you want to logout?")
+            builder.setPositiveButton("Logout") { _, _ ->
+                deleteDatabaseFile(this@HomeActivity, "app-database")
+                baseContext.deleteDatabase("app-database")
+                prefs.clear()
+                startActivity(Intent(this, LoginActivity::class.java))
+                finishAffinity()
+            }
+            builder.setNegativeButton("Cancel") { dialog, _ ->
+                dialog.cancel()
+            }
+            val dialog: AlertDialog = builder.create()
+            dialog.show()
             prefs.SP_ACCESS_TOKEN_KEY = PreferenceHelper.ACCESS_TOKEN
             prefs.SP_JWT_TOKEN_KEY = PreferenceHelper.JWT_TOKEN
-            if (nav_view.getHeaderView(0).login_button.text == "Logout") {
                 removeShortcuts()
                 invalidateToken()
-            }
             startActivity(intentFor<LoginActivity>().singleTop())
             finish()
```

Comment:
All the six lines above should execute only when the user confirms the logout.

Reply:
@aggrawalpulkit596

Reply:
@aggrawalpulkit596

Reply:
@SaumyaSingh1 firstly correct the logic and secondly there are merge heads in your code fix that too read about github in the contribution guidelines you'll know